### PR TITLE
EDLY-2158 Update user_sites api for mobile

### PR DESCRIPTION
**Pull request by iam-mhaseeb**
**Description:** EDLY-2158 Updated user_sites API for mobile

**Jira:** https://edlyio.atlassian.net/browse/EDLY-2158

This is `sample site configuration`:
Screenshot 2020-11-12 at 12 27 46 PM

This is `sample data`:
```
[
  {
    "app_config": "{\"COURSE_SHARING_ENABLED\": true, \"COURSE_VIDEOS_ENABLED\": false, \"COURSE_DATES_ENABLED\": true, \"DISCUSSIONS_ENABLED\": false, \"API_HOST_URL\": \"http://edx.devstack.lms:18000\", \"ORGANIZATION_CODE\": \"edx\"}",
    "site_data": {
      "logo": "https://edly-cloud-static-assets.s3.amazonaws.com/default-branding/logo.png",
      "logo-white": "https://edly-cloud-static-assets.s3.amazonaws.com/default-branding/logo-white.png",
      "favicon": "https://edly-cloud-static-assets.s3.amazonaws.com/default-branding/favicon.ico",
      "primary": "#3E99D4",
      "secondary": "#3E99D4",
      "display_name": "edX LMS",
      "site_url": "http://edx.devstack.lms:18000"
    },
    "mobile_enabled": true
  },
  {
    "app_config": null,
    "site_data": {
      "logo": "https://edly-cloud-static-assets.s3.amazonaws.com/red-theme/logo.png",
      "logo-white": "https://edly-cloud-static-assets.s3.amazonaws.com/red-theme/logo-white.png",
      "favicon": "https://edly-cloud-static-assets.s3.amazonaws.com/red-theme/favicon.png",
      "primary": "#dd1f25",
      "secondary": "#dd1f25",
      "display_name": "Red LMS",
      "site_url": "http://red.edx.devstack.lms:18000"
    },
    "mobile_enabled": false
  }
]
```

### CODE REVIEW

**Review by burhandodhy — APPROVED** [review]

File: openedx/features/edly/api/serializers.py [hunk 1]
```diff
@@ -37,4 +53,22 @@ def get_site_data(self, obj):
...
         return site_data
+
+    def get_mobile_enabled(self, obj):  # pylint: disable=unused-argument
+        """
+        Returns mobile_enabled flag
+        """
+        mobile_enabled = get_value_for_org(
+            self.context['edly_sub_org_of_user'].edx_organization.short_name,
+            'MOBILE_ENABLED',
+            default=False
```
Comment (farhan) [inline openedx/features/edly/api/serializers.py:72]: @zubair-arbi @iam-mhaseeb Should the mobile support be disabled by default?

File: openedx/features/edly/api/tests/test_serializers.py [hunk 2]
```diff
@@ -75,3 +86,22 @@ def test_site_data(self):
             assert color_value == serializer.data['site_data'].get(color_key)
 
         assert self.edly_sub_org_of_user.lms_site.name == serializer.data['site_data'].get('display_name')
+
+        protocol = 'https' if self.request.is_secure() else 'http'
+        url = self.test_site_configuration['SITE_NAME']
+        expected_site_url = '{}://{}'.format(protocol, url) if url else ''
+        assert expected_site_url == serializer.data['site_data'].get('site_url')
```
Comment (farhan) [inline openedx/features/edly/api/tests/test_serializers.py:93]: We should add the generated url validation as well here like
https://www.kite.com/python/answers/how-to-check-if-a-string-is-a-url-in-python

**Review by farhan — CHANGES_REQUESTED** [review]

File: openedx/features/edly/api/serializers.py [hunk 3]
```diff
@@ -1,29 +1,45 @@
...
+            )
+            url = get_value_for_org(
+                self.context['edly_sub_org_of_user'].edx_organization.short_name,
+                'SITE_NAME',
+                default=''
+            )
+            protocol = 'https' if self.context['request'].is_secure() else 'http'
+            mobile_app_config['API_HOST_URL'] = '{}://{}'.format(protocol, url) if url else ''
+            mobile_app_config['ORGANIZATION_CODE'] = self.context['edly_sub_org_of_user'].edx_organization.short_name
+            return str(json.dumps(mobile_app_config))
```
Comment (rehan99000) [inline openedx/features/edly/api/serializers.py:37]: `json.dump` should already return a string. Why typecasting again?

File: openedx/features/edly/api/serializers.py (hunk 1 above)
Comment (iam-mhaseeb) [reply openedx/features/edly/api/serializers.py:72]: If there's no `MOBILE_ENABLED` flag in site_configuraton then it will be considered as disabled automatically.

File: openedx/features/edly/api/tests/test_serializers.py (hunk 2 above)
Comment (iam-mhaseeb) [reply openedx/features/edly/api/tests/test_serializers.py:93]: Good suggestion, adding it here.

**Review by rehan99000 — CHANGES_REQUESTED** [review]
minor change and question

File: openedx/features/edly/api/serializers.py (hunk 3 above)
Comment (iam-mhaseeb) [reply openedx/features/edly/api/serializers.py:37]: @rehan99000 That was the requirement of the mobile team. They wanted JSON data wrapped in another string.
```
"app_config": "{\"COURSE_SHARING_ENABLED\": true, \"COURSE_VIDEOS_ENABLED\": false, \"COURSE_DATES_ENABLED\": true, \"DISCUSSIONS_ENABLED\": false, \"API_HOST_URL\": \"http://edx.devstack.lms:18000\", \"ORGANIZATION_CODE\": \"edx\"}",
```

File: openedx/features/edly/api/v1/urls.py [hunk 4]
```diff
@@ -3,6 +3,6 @@
 from openedx.features.edly.api.v1.views.user_sites import UserSitesViewSet
 
 router = routers.SimpleRouter()
-router.register(r'user-sites', UserSitesViewSet, base_name='user-sites')
+router.register(r'user_sites', UserSitesViewSet, base_name='user_sites')
```
Comment (rehan99000) [inline openedx/features/edly/api/v1/urls.py:6]: why this change?

Comment (iam-mhaseeb) [reply openedx/features/edly/api/v1/urls.py:6]: That was also a requirement of the mobile team.

Comment (rehan99000) [reply openedx/features/edly/api/v1/urls.py:6]: oh okay

**Review by rehan99000 — APPROVED** [review]

**Review by asadmanzoor93 — APPROVED** [review]

File: openedx/features/edly/api/tests/test_serializers.py [hunk 5]
```diff
@@ -40,27 +45,57 @@ def setUp(self):
...
+        Validates a given string as url
+        """
+        validate = URLValidator()
+        try:
+            validate(url)
+        except ValidationError:
+            return False
+        return True
+
+    def get_expected_url(self):
```
Comment (zubair-arbi) [inline openedx/features/edly/api/tests/test_serializers.py:65]: `get_expected_host_url`

**Review by zubair-arbi — APPROVED** [review]
Just one suggestion about method rename. Otherwise changes looks good. 👍 

Don't forget to rebase/squash your changes.

File: openedx/features/edly/api/tests/test_serializers.py (hunk 5 above)
Comment (iam-mhaseeb) [reply openedx/features/edly/api/tests/test_serializers.py:65]: @zubair-arbi This is generic function used for `site_url` and `host_url` both.

**Review by farhan — APPROVED** [review]
